src/util.c

```diff
@@ -104,6 +109,32 @@ void pg_md5_encrypt(const char *part1,
 		    const char *part2, size_t part2len,
 		    char *dest)
 {
+#ifdef USUAL_LIBSSL_FOR_TLS
+	EVP_MD_CTX mdctx;
+	uint8_t hash[MD5_DIGEST_LENGTH];
+
+	ERR_clear_error();
+
+	if (EVP_DigestInit(&mdctx, EVP_md5()) <= 0)
+	{
+		log_error("MD5 authentication: %s", ERR_reason_error_string(ERR_get_error()));
+	}
+
+	else if (EVP_DigestUpdate(&mdctx, part1, strlen(part1)) <= 0 ||
+			EVP_DigestUpdate(&mdctx, part2, part2len) <= 0 ||
+			EVP_DigestFinal_ex(&mdctx, hash, 0) <= 0 )
+	{
+		log_error("MD5 authentication: %s", ERR_reason_error_string(ERR_get_error()));
```

Comment:
Now that we can have errors when "encrypting" with md5, we should make pg_md5_encrypt return true/false accordingly. So we can correctly handle an error. We wouldn't want an error to result in comparing to an empty hash or something.